src/Illuminate/Routing/UrlGenerator.php

```diff
@@ -355,7 +355,7 @@ public function temporarySignedRoute($name, $expiration, $parameters = [], $abso
      */
     public function hasValidSignature(Request $request, $absolute = true)
     {
-        $url = $absolute ? $request->url() : '/'.$request->path();
+        $url = $absolute ? url()->current() : '/'.$request->path();
```

Comment:
We can't use this helper because it's only available in the foundation package surerly?

Reply:
Works perfectly within my environment, generates the correct URL and respects protocols. Maybe this is working because of dependency injection from the underlying framework?

Reply:
It works perfectly because you're loading in the entire framework. People using the package individual will experience problems. Since you're already in the url generator you can simply do `$this->current()` I believe.

Reply:
that works prefectly @driesvints I've committed it back.